[WPE] Unreviewed test gardening

kate-k-lee opened the pull request:
#### cf006175b05518409b0fe093ed32904ecea338de
<pre>
[WPE] Unreviewed test gardening

Updated the status of flaky tests.

* LayoutTests/platform/wpe/TestExpectations:

Canonical link: <a href="https://commits.webkit.org/302973@main">https://commits.webkit.org/302973@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/6d4b7ba536d173fd54cad1790349279096ac7490

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/130042 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/2303 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/40903 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/137437 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/81558 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/2268 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/2195 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/99060 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/66859 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/132989 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/1696 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/116474 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/79761 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/1625 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/34600 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/80709 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/110138 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/35109 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/139916 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/2097 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/1944 "Found 1 new test failure: imported/w3c/web-platform-tests/webvtt/rendering/cues-with-video/processing-model/selectors/cue/white-space_normal_wrapped.html (failure)") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/107564 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/2142 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/112821 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/107457 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/1674 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/31269 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/54931 "The change is no longer eligible for processing.") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/20362 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/2170 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/1986 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/2019 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/2093 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

dpino commented on LayoutTests/platform/wpe/TestExpectations:
```diff
@@ -575,9 +571,7 @@ webkit.org/b/301555 transforms/2d/zoom-menulist.html [ Skip ]
-webkit.org/b/119253 fast/workers/dedicated-worker-lifecycle.html [ Timeout Pass ]
-
-webkit.org/b/129758 js/dom/create-lots-of-workers.html [ Timeout Pass ]
+webkit.org/b/119253 fast/workers/dedicated-worker-lifecycle.html [ Pass ]
```
By default, tests are expected to pass. We only mark a test as `Pass`, if the test is expected to fail.

For instance, if in `LayoutTests/TestExpectations` there was an entry like:

```
fast/workers/dedicated-worker-lifecycle.html [ Failure ]
```

then it would be OK to overwrite this result with Pass, if the test is passing in WPE.

Since `fast/workers/dedicated-worker-lifecycle.html` is not present in `LayoutTests/TestExpectations`, that means the test is expected to Pass, thus this entry is redundant.

dpino commented on LayoutTests/platform/wpe/TestExpectations:
```diff
@@ -179,7 +179,6 @@ webkit.org/b/212083 loader/navigation-policy/should-open-external-urls/user-gest
 
 # Mediastream
 webkit.org/b/213202 fast/mediastream/getUserMedia-grant-persistency3.html [ Failure ]
-http/wpt/mediastream/mediastreamtrackprocessor-videoframe-timestamp.html [ Pass Slow Failure ]
```
Looking up this test in results.webkit.org, shows the test is still flakey in WPEWebKit and WebKitGTK:

https://results.webkit.org/?suite=layout-tests&test=http%2Fwpt%2Fmediastream%2Fmediastreamtrackprocessor-videoframe-timestamp.html&platform=WPE&platform=GTK

dpino commented on LayoutTests/platform/wpe/TestExpectations:
```diff
@@ -604,22 +598,17 @@ webkit.org/b/235277 [ Debug ] imported/w3c/web-platform-tests/html/canvas/offscr
-webkit.org/b/252878 fast/hidpi/image-set-units.html [ ImageOnlyFailure Pass ]
-webkit.org/b/252878 [ Release ] fast/scrolling/keyboard-scrolling-distance-pageDown.html [ Pass Timeout ]
+webkit.org/b/252878 fast/hidpi/image-set-units.html [ Timeout Pass ]
+webkit.org/b/252878 [ Release ] fast/scrolling/keyboard-scrolling-distance-pageDown.html [ Pass ]
```
This is a good use of Pass. The test is meant to be skipped in all WebKit ports, but it's passing in WPE.

dpino commented on LayoutTests/platform/wpe/TestExpectations:
```diff
@@ -604,22 +598,17 @@ webkit.org/b/235277 [ Debug ] imported/w3c/web-platform-tests/html/canvas/offscr
+webkit.org/b/252878 http/wpt/cross-origin-opener-policy/header-parsing-with-report-to.https.html [ Failure Pass Timeout ]
 webkit.org/b/252878 http/wpt/service-workers/controlled-sharedworker.https.html [ Failure Pass ]
-webkit.org/b/252878 imported/blink/fast/hidpi/border-background-align.html [ Crash ImageOnlyFailure Pass ]
+webkit.org/b/252878 imported/blink/fast/hidpi/border-background-align.html [ Pass ]
```
This entry is redundant since the test is expected to pass.

dpino commented on LayoutTests/platform/wpe/TestExpectations:
```diff
@@ -1033,10 +1011,8 @@ webgl/2.0.y/conformance/offscreencanvas/offscreencanvas-transfer-image-bitmap.ht
 fast/url/url-hostname-encoding.html [ Failure ]
-  
-imported/w3c/web-platform-tests/websockets/interfaces/WebSocket/constants/001.html?wss [ Pass Timeout ]
+
```
Nit: remove extra line.

dpino reviewed (approved):
Overall it looks good to me.

For those tests marked as Pass only, I'd suggest to double-check the entry is actually needed (justified if test is skipped or it's expected to fail).

kate-k-lee committed:
[WPE] Unreviewed test gardening

Updated the status of flaky tests.

* LayoutTests/platform/wpe/TestExpectations:

Canonical link: https://commits.webkit.org/302973@main
